README.md

```diff
@@ -1516,6 +1517,132 @@ FunctionDeclaration(path) {
 
 ----
 
+# Building Nodes
+
+When writing transformations you'll often want to build up some nodes to insert
+into the AST. As mentioned previously, you can do this using the
+[builder](#builder) methods in the [`babel-types`](#babel-types) package.
+
+The method signature for a builder is simple, the name is the same as the node
+type you want to build, but with the first letter lowercase. For example if you
+wanted to build a `MemberExpression` you would use `t.memberExpression(...)`.
+
+The arguments of these builders is decided by the node definition. There's some
+work that's being done to generate easy-to-read documentation on the
+definitions, but for now they can all be found
+[here](https://github.com/babel/babel/tree/master/packages/babel-types/src/definitions).
+
+A node definition looks like the following:
+
+```js
+defineType("MemberExpression", {
+  builder: ["object", "property", "computed"],
```

Comment:
There is also a case with there are more fields/properties to fill in than actual elements in the builder array because it might be too long of a method signature so maybe mention that.

Example is with a `ClassMethod` https://github.com/babel/babel/blob/master/packages/babel-types/src/definitions/es2015.js#L238-L276

The builder doesn't have a generator/async field in the builder array so you would have to add it afterwards.